### PR TITLE
Fix `io.cp2k.input.DataFile`

### PR DESCRIPTION
**Pull request by DanielYang59**
## Summary

Fixed a flawed change in `io.cp2k.input.DataFile.from_file` made by me in #3705, discussed in https://github.com/materialsproject/pymatgen/pull/3705#discussion_r1558895539. Also added comment to prevent this from happening again.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the file loading process in the `DataFile` class to improve data handling and assignment.
- **Tests**
	- Updated assertions in testing for partial charges to improve clarity and consistency.
- **Refactor**
	- Improved variable naming consistency in the `plot_power_factor_mu`, `plot_power_factor_temp`, and `plot_power_factor_dop` functions.
- **Tests**
	- Made various variable name changes and added imports in test files to enhance clarity and maintain consistency.

### CODE REVIEW

**Comment by DanielYang59** [comment]
Can you please review this fix? @janosh Apologize again for the trouble.

**Review by janosh — APPROVED** [review]
thanks for this fix. i added several new tests and made the `DataFile.from_file` an `abstractmehod`

**Comment by DanielYang59** [comment]
Thanks for reviewing and making it a real "fix" 😄

File: tests/io/test_openff.py [hunk 1]
```diff
@@ -191,7 +190,7 @@ def test_assign_partial_charges(mol_files):
     openff_mol, atom_map = add_conformer(openff_mol, geometry)
     partial_charges = np.load(mol_files["CCO_charges"])
     openff_mol = assign_partial_charges(openff_mol, atom_map, "am1bcc", partial_charges)
-    assert np.allclose(openff_mol.partial_charges.magnitude, partial_charges)
+    assert_allclose(openff_mol.partial_charges.magnitude, partial_charges)
```
Comment (DanielYang59) [inline tests/io/test_openff.py:193]: Wondering what is the difference here? (thought the original syntax is more consistent with the `assert condition` syntax)

From the [assert_allclose documentation](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html), these two seems equivalent:
>The test is equivalent to allclose(actual, desired, rtol, atol) (note that allclose has different default values). It compares the difference between actual and desired to atol + rtol * abs(desired).

Comment (janosh) [reply tests/io/test_openff.py:193]: compare the error messages when they each fail. `assert_allclose` gives a much more informative message

Comment (DanielYang59) [reply tests/io/test_openff.py:193]: Thanks for the input! Appreciate that.

**Pull request merged**